enterprise/forms.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+"""
+Forms to be used in the enterprise djangoapp
+"""
+from django import forms
+from enterprise.models import EnterpriseCustomer
+from enterprise.utils import get_all_catalogs
+
+class EnterpriseCustomerAdminForm(forms.ModelForm):
+    """
+    Alternate form for the EnterpriseCustomer admin page; fetches
+    catalog names and IDs from the Course Discovery API
+    """
+    class Meta:
+        model = EnterpriseCustomer
+        fields = (
+            'name',
+            'site',
+            'active',
+            'catalog',
+        )
+
+    def __init__(self, *args, **kwargs):
+        """
+        Save the user that the form was constructed for and initalize
+        using the parent class methods
+        """
+        super(EnterpriseCustomerAdminForm, self).__init__(*args, **kwargs)
+        self.fields['catalog'] = forms.ChoiceField(choices=self.get_catalog_options())
```

Comment:
@haikuginger nit/optional suggestion: Just to make sure you haven't missed that approach - you could override field widget using `Meta.widgets` attribute. But probably you can't avoid using `self` here or down the stream, so if it's the case I'm fine with this way of doing it.